SafeguardSessions/test/Unit/schema/TestFlatten.query.pq

```diff
@@ -4,8 +4,84 @@ SchemaUtils = Extension.ImportModule("SchemaUtils.pqm");
 
 SchemaUtils.FieldNamesOfType = SchemaUtils[FieldNamesOfType];
 SchemaUtils.IsFlat = SchemaUtils[IsFlat];
+SchemaUtils.FlattenLists = SchemaUtils[FlattenLists];
+SchemaUtils.FlattenRecords = SchemaUtils[FlattenRecords];
 SchemaUtils.Flatten = SchemaUtils[Flatten];
 
+TestFlatteningLists = () =>
+    let
+        AssertResult = (description as text, t as table, fieldNames as list, expectedTable as table) =>
+            Fact(description, expectedTable, SchemaUtils.FlattenLists(t, fieldNames)),
+        fieldNames = {"list_field"},
+        cases = {
+            {
+                "List field is flattened correctly",
+                #table({"list_field"}, {{{"1", "2"}}, {{"3", "4"}}}),
+                fieldNames,
+                #table(type table [list_field = any], {{"1,2"}, {"3,4"}})
+            },
+            {
+                "List field with empty list is flattened correctly",
+                #table({"list_field"}, {{{}}, {{"3", "4"}}}),
+                fieldNames,
+                #table(type table [list_field = any], {{""}, {"3,4"}})
+            },
+            {
+                "List field with null value is flattened correctly",

```

Comment:
Can you add a scenario, where the list contains a null value and a non-value as well?

Reply:
What do you mean by a non-value? Text.Combine can only handle text values.

Should we handle other types as well? Plus 1 step, but it is doable.

Reply:
No, good point. As long as we do not need to handle lists that includes items other than string it is good.

However, because of this restriction, I think we should check if the list contains non string items in which case we should log the field name and the whole list and raise a dedicated error, so that we don't spend hours finding this bug when the REST would return a list field including numbers for example.

Reply:
Working on it.

Reply:
Done!